api/admin/controller/__init__.py

```diff
@@ -2483,3 +2501,23 @@ def _is_email(self, email):
         """Email addresses must be in the format 'x@y.z'."""
         email_format = ".+\@.+\..+"
         return re.search(email_format, email)
+
+    def validate_url(self, settings):
+        """Find any URLs that the user has submitted, and make sure that
+        they are in a valid format."""
+        settings = settings or self._get_settings()
+        if isinstance(settings, (list,)):
+            # Find the fields that have to do with URLs
+            url_fields = filter(lambda s: (s.get("format") == "url" or s.get("key") == "url") and flask.request.form.get(s.get("key")), settings)
+            # Narrow the URL-related fields down to the ones for which the user actually entered a value
+            url_inputs = [flask.request.form.get(field.get("key")) for field in url_fields]
+        else:
+            # The SitewideSettingsController passes in a string
+            url_inputs = [settings]
+        for url in url_inputs:
+            if not self._is_url(url):
+                return INVALID_URL.detailed(_('"%(url)s" is not a valid URL.', url=url))
+
+    def _is_url(self, url):
+        url_format = "(http[s]?://)+"
+        return re.search(url_format, url)
```

Comment:
I think it would be better to remove the regular expression altogether and do something like this:

```
return any([url.startswith(protocol + "://") for protocol in "http", "https"])
```

Because, the other problem I was going to mention -- and this may be the cause of the 'localhost' problem you were having -- is that the regular expression isn't tied to the front of the string. This regular expression will match any string that _contains_ a URL, such as `Hi, Rhonda, the URL to the library's website is http://library.org/. Be sure to get it in the circulation manager ASAP! Thanks!"

To fix this in a regular expression you'd stick a carat on the start of the string, but given that I also asked you to dramatically simplify the regular expression I think using `startswith` is simpler and more explicit.